### PR TITLE
Fix minor issues in Makefile for Release 0.1 (#33)

### LINKED ISSUE #33: Makefile revamping; Fixing the issues pointed in Release 0.1
image
image
image
image

### PR DESCRIPTION
**Pull request by Tabrez-dev**
This PR addresses minor issues in the Makefile for Release 0.1, as mentioned in issue #33. 
This PR includes fixes for concerns raised in the issue except 2nd one:
1) Removed GCC major minor variables
3) Added min version for newer flags
4) Removed all Phony variables.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,10 +1,23 @@
...
+endef
+
+GCC_MIN_VERSION := 11.0.0
+GCC_VERSION := $(shell riscv64-unknown-elf-gcc --version | head -n 1 | sed 's/.* //')  # For example: 9.3.0
+
+MIN_VERSION_NUMERIC := $(call normalize_version, $(GCC_MIN_VERSION)) # Converts 11.0.0 -> 110000
+CURRENT_VERSION_NUMERIC := $(call normalize_version, $(GCC_VERSION))  # Converts 9.3.0 -> 90300
+
+# Compare normalized versions and issue a warning if the current version is lower than the minimum required version
+#I dont know about you guys but my gcc version is 9.3.0 and i cant seem to go higher so I will get a warning
```
Comment (pranjalchanda08) [inline Makefile]: can you remove this comment

File: Makefile [hunk 2]
```diff
@@ -1,10 +1,23 @@
...
+ifeq ($(shell [ $(CURRENT_VERSION_NUMERIC) -lt $(MIN_VERSION_NUMERIC) ] && echo 1 || echo 0), 1)
+  $(warning Bad toolchain version $(GCC_VERSION). Minimum required: $(GCC_MIN_VERSION))
```
Comment (pranjalchanda08) [inline Makefile]: This shall be provided as **Recommended** not as a must. As we are still able to support the toolchain version.

**Review by pranjalchanda08 — CHANGES_REQUESTED** [review]

File: Makefile (hunk 2 above)
Comment (Tabrez-dev) [reply Makefile]: Should i remove this check? or change it like this:

```
# Compare normalized versions and issue a warning if the current version is lower than the minimum required version
ifeq ($(shell [ $(CURRENT_VERSION_NUMERIC) -lt $(MIN_VERSION_NUMERIC) ] && echo 1 || echo 0), 1)
	$(warning Bad toolchain version $(GCC_VERSION). Recommended: $(GCC_MIN_VERSION))
endif
```

File: Makefile (hunk 1 above)
Comment (Tabrez-dev) [reply Makefile]: Done

File: Makefile (hunk 2 above)
Comment (pranjalchanda08) [reply Makefile]: warning is ok

**Review by pranjalchanda08 — APPROVED** [review]

**Comment by pranjalchanda08** [comment]
Hi @Tabrez-dev is this PR part of #37 as well? If yes please close this

**Comment by Tabrez-dev** [comment]
Major make files fix branch fixed all the issues of #33

**Pull request closed without merging**